### ISSUE TITLE
Resolving Dns in Parallel increases API latency

### ISSUE THREAD

**Issue by Krishnasony** [dnsjava/dnsjava#362]
Need Help: I have implemented parallel DNS to fix the increase in API latency previously I was using sequential which increased API latency. However, implementing Parallel DNS increased further instead of getting it fixed.
```
class ParallelDns(
  private val dnsList: List<Dns>,
  private val analytics: DnsAnalytics,
  coroutineContext: CoroutineContext,
) : Dns, CoroutineScope by CoroutineScope(coroutineContext) {

  @Throws(UnknownHostException::class)
  override fun lookup(hostname: String): List<InetAddress> {
    analytics.logResolvingDns(hostname)
    val deferredResults = dnsList.map { dns ->
      async(ioDispatcher) {
        try {
          val addresses = dns.lookup(hostname)
          if (addresses.isNotEmpty()) {
            logger().info { "DNS resolved for host $hostname using ${dns.javaClass.simpleName}" }
            analytics.logDnsResolved(hostname)
            addresses
          } else {
            throw UnknownHostException("Empty result from ${dns.javaClass.simpleName}")
          }
        } catch (e: Exception) {
          val errorMessage = "Failed to resolve $hostname using ${dns.javaClass.simpleName}: $e"
          logger().error { errorMessage }
          val exception = AddressRecordNotFoundException(cause = e, message = errorMessage)
          recordException(exception)
          analytics.logFailedToResolveHost(hostname)
          null
        }
      }
    }

    // Wait for the first successful response
    return runBlocking { maybeGetAddresses(hostname, deferredResults) }
  }

  private suspend fun maybeGetAddresses(
    hostname: String,
    deferredResults: List<Deferred<List<InetAddress>?>>,
  ): List<InetAddress> =
    deferredResults.firstNotNullOfOrNull { it.await() }?.also {
      logger().info { "DNS resolved successfully for $hostname" }
    } ?: run {
      logger().error { "Failed to resolve $hostname using all DNS resolvers" }
      analytics.logFailedToResolveHostUsingAnyDns(hostname)
      throw UnknownHostException("Failed to resolve $hostname using any DNS")
    }
}
```

Here are the resolvers I'm using
```
{
  "dns_resolvers": [
    {
      "name": "cloudflare",
      "resolvers": [
        "1.1.1.1",
        "1.0.0.1",
        "2606:4700:4700::1111",
        "2606:4700:4700::1001"
      ]
    },
    {
      "name": "google",
      "resolvers": [
        "8.8.8.8",
        "8.8.4.4",
        "2001:4860:4860::8888",
        "2001:4860:4860::8844"
      ]
    },
    {
      "name": "quad9",
      "resolvers": [
        "9.9.9.9",
        "149.112.112.112",
        "2620:fe::fe",
        "2620:fe::9"
      ]
    },
    {
      "name": "opendns",
      "resolvers": [
        "208.67.222.222",
        "208.67.220.220",
        "2620:119:35::35",
        "2620:119:53::53"
      ]
    }
  ]
}
```

**Comment by ibauersachs** [comment]
I have no idea what you're doing here, but I don't see a single class name from dnsjava.
